Add command gui

chloeelim opened the pull request:
- [X] `addcus` GUI Window
- [X] `addcom` GUI Window
- [X] `additer` GUI Window

### `addcus` GUI Window (`AddCustomerWindow`)

https://user-images.githubusercontent.com/79991214/197326198-52d0e6cc-805f-45bb-91ce-2b4bcfd6c827.mov

**Design considerations**
Inherits from the `CustomerWindow`, which stores all the Customer fields. The `CustomerWindow` is abstracted out to make adding an `editcus` window easier as the `EditCustomerWindow` can simply inherit from the `CustomerWindow`  and implement its on handling of the edit command/ pre-fill in the respective input fields.


## `addcom` GUI Window (`AddCommissionWindow`)

https://user-images.githubusercontent.com/79991214/197326192-0547fd0a-8849-44c2-9172-24abfe8c5ac1.mov

**Design considerations**
Like the `CustomerWindow`, the `AddCommissionWindow` also inherits from a `CommissionWindow` to support easier extension of `EditCommissionWindow` in later enhancements.

## `additer` GUI Window (`AddIterationWindow`)

https://user-images.githubusercontent.com/79991214/197326385-a17b436a-7773-4fbc-a436-ff2d2c690768.mov

https://user-images.githubusercontent.com/79991214/197326504-4d907dcf-27be-4fdd-b858-445b4e633a8c.mov

Supports uploading of images by file selection via the file explorer, or via drag and drop.


**Design considerations**
Likewise, `AddIterationWindow` also inherits from an `IterationWindow` to support easier extension of `EditIterationWindow` in later enhancements.

---

Notes:
1. this PR also enforces a max character cap of 25 characters on all tags because very long tag names can cause the GUI to look buggy.
<img width="1082" alt="Screenshot 2022-10-20 at 3 19 20 PM" src="https://user-images.githubusercontent.com/79991214/197326888-5528ea82-5e52-4c0e-972e-d11705db60a2.png">

2. in addition, the PR also fixes a bug in selecting a blank image path when creating an iteration, where a `NullPointerException` is being thrown as the file type of the blank image path cannot be determined (try `additer n/test d/2022-10-22 f/abc p/`) [#120]. See commit c111f4f291dca32b4f1b33dfe08cb5639394b46f.

nvmdava commented:
Really great updates 😄 . Just a few things I want to point out.

When there is an error in the customer add, the AddCustomer button seems to half-disappear on the bottom. Can we increase the height?
![image](https://user-images.githubusercontent.com/86460216/197410079-9498c880-ccee-453a-b891-d0d5f6bdc4db.png)
Also maybe can have an image in the top-left corner icon :P.

On the commissions pop up, would it be possible to instead have Completed as a checkbox, or a radio button?
![image](https://user-images.githubusercontent.com/86460216/197410285-2f7a71bb-5d24-429a-9454-087a7a219967.png)

But I am quite amazed how neatly you've done the tags part.
Oh and finally, on AddIteration part, can we have the image not stretched out?(can scale, but keep proportion.)
![image](https://user-images.githubusercontent.com/86460216/197410415-108c62bd-8b3f-45ab-be7c-bb2e7c6a17ef.png)

Overall very nice works 💯💯💯

chloeelim commented:
Thanks for bring these up!

Interestingly the CustomerWindow appears to be ok on my computer- might be a OS dependent issue (gg). I've increased the height of the CustomerWindow and made the windows scrollable to just in case this issue crops up for others.

I also added the same art-palette icon to the windows but they don't appear for me. I think this is another OS thing (I'm using mac), could you help me check if the OS is rendering correctly for you?

I also changed the image so it takes the proportions of the selected image- should hopefully work fine:

https://user-images.githubusercontent.com/79991214/197662948-9deebd5f-cd36-40d0-b85a-b4db143a72d5.mov

And changed the completion-input to a toggle button:

<img width="715" alt="Screenshot 2022-10-25 at 9 47 44 AM" src="https://user-images.githubusercontent.com/79991214/197663036-24d95761-e03f-416d-a0b9-03b9fc29cdf1.png">

Let me know if I missed anything out!

nvmdava reviewed (approved):
Perfect, looks like everything has been attended so well. Thank you
